library/src/blas1/rocblas_rotm.hpp

```diff
@@ -0,0 +1,168 @@
+                                     rocblas_stride stride_y,
+                                     const T*       param,
+                                     rocblas_stride stride_param,
+                                     rocblas_int    batch_count,
+                                     T*             mem)
+{
+    // Memory queries must be in template as _impl doesn't have stride_param parameter (for calls from
+    // outside of rocblas)
+    if(handle->is_device_memory_size_query())
+    {
+        // TODO: Decide if we want to support this or not.
+        if(stride_param && rocblas_pointer_mode_host == handle->pointer_mode && n > 0 && incx > 0
+           && incy > 0 && batch_count > 0)
+            return handle->set_optimal_device_memory_size(sizeof(T) * batch_count * stride_param);
+        else
+            return rocblas_status_size_unchanged;
+    }
+
+    // Quick return if possible
+    if(n <= 0 || incx <= 0 || incy <= 0 || batch_count <= 0)
+        return rocblas_status_success;
+    if(rocblas_pointer_mode_host == handle->pointer_mode && param[0] == -2)
+        return rocblas_status_success;
+
+    dim3        blocks((n - 1) / NB + 1, batch_count);
+    dim3        threads(NB);
+    hipStream_t rocblas_stream = handle->rocblas_stream;
+
+    if(rocblas_pointer_mode_device == handle->pointer_mode)
+        hipLaunchKernelGGL(rotm_kernel,
+                           blocks,
+                           threads,
+                           0,
+                           rocblas_stream,
+                           n,
+                           x,
+                           offset_x,
+                           incx,
+                           stride_x,
+                           y,
+                           offset_y,
+                           incy,
+                           stride_y,
+                           param,
+                           param + 1,
+                           param + 2,
+                           param + 3,
+                           param + 4,
+                           stride_param);
+    else if(!stride_param) // single param on host
+        hipLaunchKernelGGL(rotm_kernel,
+                           blocks,
+                           threads,
+                           0,
+                           rocblas_stream,
+                           n,
+                           x,
+                           offset_x,
+                           incx,
+                           stride_x,
+                           y,
+                           offset_y,
+                           incy,
+                           stride_y,
+                           param[0],
+                           param[1],
+                           param[2],
+                           param[3],
+                           param[4],
+                           0);
+    else // array of params on host, copy to device
+    {
+        // This should NOT happen from calls from the API currently.
```

Comment:
I vote we don't support this.

Reply:
This has been addressed in 474e196. We now support:
- batched `param` for rotm(g)_(strided_)batched.
- `param` can be located on either host or device for original rotm(g)
- `param` can only be located on device for (strided_)batched versions.